<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="viewport" content="width=device-width, initial-scale=1"><title>WP_003512336.1 | Report</title><meta name="description" content="SWISS-MODEL modelling results for WP_003512336.1." /><style>
			.sequence { font-size: 12px; font-family:"Courier New", Courier, monospace;}
			h2 {font-size: 1.2em; font-family:Arial; margin:20 0 0 0}
			h3 {font-size: 1.0em; font-family:Arial; margin:0 0 0 0}
			
			.borderedTbl>tbody>tr>td, .borderedTbl th{ 
        text-align:center;
  			border-top:1px solid #DDD;
  			border-left:1px solid #DDD;
      }
      .borderedTbl{
        border-bottom:1px solid #DDD;
        border-right:1px solid #DDD; 
      }
      td { 
        -webkit-hyphens: auto;
           -moz-hyphens: auto;
                hyphens: auto; 
              }

			body {
			width:100%; 
			height:100%; 
			background-color: #E7E8EA;
			margin: 0;
			padding: 0;
			word-wrap: break-word;
			font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
      font-size: 10pt;

			}
			#content{
			padding:20px;
			width:780px;
			margin-left:auto;
			margin-right:auto;
			background-color:#FFF;}
			
			#sib_footer {
			background-color: #333333;
			color: #F7F8FA;
			font-family: Verdana;
			font-size: 10px;
			font-weight: normal;
			height: 20px;
			letter-spacing: 1px;
			
			width:760px;
      margin-left:auto;
      margin-right:auto;
      padding:0px 20px 0px 20px
			}
			
			#sib_footer a {
  			color: #FFFFFF;
  			font-family: Verdana;
  			font-size: 10px;
  			font-weight: normal;
  			letter-spacing: 1px;
			}
			a, a:link, a:visited {
			color: #D6161D;
			text-decoration: none;
			} 	

			.alTbl{ font-family:'Courier New', Courier, monospace; }
	   img{ border:0; }
			
	@media print{
	
	#content{
			padding:20px;
			background-color:#FFF;
		}
	body{ 
	    width:100%;
	    height:100%; 
	    background-color: #FFF;
			padding: 0;
		}	
	#sib_footer{
		display:none;
	}
	}
      
    </style></head><body><div id="content"><table border=0 width="100%"><td valign="middle"><a title="SIB Swiss Institute of Bioinformatics" href="https://www.sib.swiss"><img src="images/sib_ExPASy_logo_small.png"></a></td><td valign="middle"><a title="Biozentrum, University of Basel" href="https://www.biozentrum.unibas.ch"><img src="images/biozentrum_en.png"></a></td><td valign="middle"><div style="font-size:1.9em;">SWISS-MODEL Homology Modelling Report</div></td></tr></table><div style="clear:both"></div><h2>Model Building Report</h2><p>
This document lists the results for the homology modelling project &quot;WP_003512336.1&quot; submitted to SWISS-MODEL workspace
on May 12, 2019, 8:37 p.m..The submitted primary amino acid sequence is given in Table T1.</p><p>If you use any results in your research, please cite the relevant publications:</p><ul><li>Waterhouse, A., Bertoni, M., Bienert, S., Studer, G., Tauriello, G., Gumienny, R.,
      Heer, F.T., de Beer, T.A.P., Rempfer, C., Bordoli, L., Lepore, R., Schwede, T.   
      SWISS-MODEL: homology modelling of protein structures and complexes.
      Nucleic Acids Res. 46(W1), W296-W303 (2018).
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/29788355"
       target="_blank"><img src="images/pubmed.jpg
                                 "
                            width="17" height="16" align="absmiddle"></a><a href="https://dx.doi.org/10.1093/nar/gky427"
       target="_blank"><img src="images/doi.jpg
                                 "
                            width="30" height="16" align="absmiddle"
                            style="padding-left:2px"></a></li><li><a name="promodII"></a>Guex, N., Peitsch, M.C., Schwede, T. Automated
          comparative protein structure modeling with SWISS-MODEL and
          Swiss-PdbViewer: A historical perspective. Electrophoresis 30,
          S162-S173 (2009).
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/19517507"
       target="_blank"><img src="images/pubmed.jpg
                                 "
                            width="17" height="16" align="absmiddle"></a><a href="https://dx.doi.org/10.1002/elps.200900140"
       target="_blank"><img src="images/doi.jpg
                                 "
                            width="30" height="16" align="absmiddle"
                            style="padding-left:2px"></a></li><li>Bienert, S., Waterhouse, A., de Beer, T.A.P., Tauriello, G., Studer,
    G., Bordoli, L., Schwede, T. The SWISS-MODEL Repository - new features and
    functionality. Nucleic Acids Res. 45, D313-D319 (2017).
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/27899672"
       target="_blank"><img src="images/pubmed.jpg
                                 "
                            width="17" height="16" align="absmiddle"></a><a href="https://dx.doi.org/10.1093/nar/gkw1132"
       target="_blank"><img src="images/doi.jpg
                                 "
                            width="30" height="16" align="absmiddle"
                            style="padding-left:2px"></a></li><li><a name="qmean"></a>Benkert, P., Biasini, M., Schwede, T. Toward the
        estimation of the absolute quality of individual protein structure
        models. Bioinformatics 27,
        343-350 (2011).
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/21134891"
       target="_blank"><img src="images/pubmed.jpg
                                 "
                            width="17" height="16" align="absmiddle"></a><a href="https://dx.doi.org/10.1093/bioinformatics/btq662"
       target="_blank"><img src="images/doi.jpg
                                 "
                            width="30" height="16" align="absmiddle"
                            style="padding-left:2px"></a></li><li><a name="qsqe"></a>Bertoni, M., Kiefer, F., Biasini, M., Bordoli, L.,
        Schwede, T. Modeling protein quaternary structure of homo- and
        hetero-oligomers beyond binary interactions by homology. Scientific
        Reports 7 (2017).
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/28874689"
       target="_blank"><img src="images/pubmed.jpg
                                 "
                            width="17" height="16" align="absmiddle"></a><a href="https://dx.doi.org/10.1038/s41598-017-09654-8"
       target="_blank"><img src="images/doi.jpg
                                 "
                            width="30" height="16" align="absmiddle"
                            style="padding-left:2px"></a></li></ul></p><h2>Results</h2><p>
The SWISS-MODEL template library (SMTL version 2019-05-08, PDB release 2019-05-03) was searched with 
BLAST (<a href="#blastp">Camacho et al.</a>) and HHBlits (<a href="#hhblits">Remmert et al.</a>) 

 for evolutionary related structures matching the target sequence in Table T1. For details on the template search, see Materials and Methods. Overall 28 templates were found (Table T2).</p><h2>Models</h2><p>No models have been built for this project.</p><h2>Materials and Methods</h2><h2>Template Search</h2><p>Template search with BLAST and HHBlits
 has been performed against the SWISS-MODEL template library (SMTL, last update: 2019-05-08, last included PDB release: 2019-05-03).</p><p>
The target sequence was searched with BLAST against the primary amino acid sequence contained in the SMTL.


A total of 5 templates were found.


</p><p>
An initial HHblits profile has been built using the procedure outlined in (Remmert et al.), followed by 1 iteration of HHblits against NR20. The obtained profile has then be searched against all profiles of the SMTL. A total of 23 templates were found.</p><h2>Model Building</h2><p>
Models are built based on the target-template alignment using ProMod3. Coordinates which are conserved between the target and the template are copied from the template to the model. Insertions and deletions are remodelled using a fragment library. Side chains are then rebuilt. Finally, the geometry of the resulting model is regularized by using a force field. In case loop modelling with ProMod3 fails, an alternative model is built with PROMOD-II (<a href="#promodII">Guex et al.</a>).
</p><h2>Model Quality Estimation</h2><p>
The global and per-residue model quality has been assessed using the QMEAN scoring function (<a href="#qmean">Benkert et al.</a>) . For improved performance, weights of the individual QMEAN terms have been trained specifically for SWISS-MODEL. 
</p><h2>Ligand Modelling</h2><p>
Ligands present in the template structure are transferred by homology to the model when the following criteria are met: (a) The ligands are annotated as biologically relevant in the template library, (b) the ligand is in contact with the model, (c) the ligand is not clashing with the protein, (d) the residues in contact with the ligand are conserved between the target and the template. If any of these four criteria is not satisfied, a certain ligand will not be included in the model. The model summary includes information on why and which ligand has not been included.</p><h2>Oligomeric State Conservation</h2><p>
The quaternary structure annotation of the template is used to model the target sequence in its oligomeric form. The method (<a href="#qsqe">Bertoni et al.</a>) is based on a supervised machine learning algorithm, Support Vector Machines (SVM), which combines interface conservation, structural clustering, and other template features to provide a quaternary structure quality estimate (QSQE). The QSQE score is a number between 0 and 1, reflecting the expected accuracy of the interchain contacts for a model built based a given alignment and template. Higher numbers indicate higher reliability. This complements the GMQE score which estimates the accuracy of the tertiary structure of the resulting model.</p><h2>References</h2><p><ul><li><a name="blastp"></a><b>BLAST</b><br>
        Camacho, C., Coulouris, G., Avagyan, V., Ma, N., Papadopoulos, J.,
        Bealer, K., Madden, T.L. BLAST+: architecture and applications. BMC
        Bioinformatics 10, 421-430 (2009).
        <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/20003500"><img width="17" align="absmiddle" height="16"
               src="images/pubmed.jpg
                    "></a><a href="https://dx.doi.org/10.1186/1471-2105-10-421"
           target="_blank"><img src="images/doi.jpg
                                     " width="30"
                                height="16" align="absmiddle"
                                style="padding-left:2px"></a></li><li><a name="hhblits"></a><b>HHblits</b><br>
        Remmert, M., Biegert, A., Hauser, A., S&ouml;ding, J. HHblits:
        lightning-fast iterative protein sequence searching by HMM-HMM
        alignment. Nat Methods 9, 173-175 (2012).
        <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/22198341"><img width="17" align="absmiddle" height="16"
               src="images/pubmed.jpg
                    "></a><a href="https://dx.doi.org/10.1038/nmeth.1818"
           target="_blank"><img src="images/doi.jpg
                                     " width="30"
                                height="16" align="absmiddle"
                                style="padding-left:2px"></a></li></ul></p><h2>Table T1:</h2><p>Primary amino acid sequence for which templates were searched and models were built.</p><div class="sequence">MAIIPSVTEKCTGAMLASAIGDALGWPNELGAKNSIKNTKVNDQFIEWTRRSGGRYWPYYEKILPGEYSDDTQMILAVARSVITGNWESTLAFKEIPFWL<br>KYERGGGSALKEAAKLLKKGIVPWEARENNKYFMAGGNGAAMRILPHVIHKQNIDLLILDVIKDCLYTHGHPRAILGASCYAYALYYLRNKKDVLEYGEL<br>IRAVIDGLKYWTTLNALPAQWLVLAEKHSGFDYLQIWFNTISNMAQKLEYIFYSLDKGLLINDDDVLKYLGCFDETSGAGDVAILSALYLASKYANNPML<br>GIKIAANLHNADTDTIASITGGLLGMLCGTEWIPFEWRLVQDYDCLVNITQILLADNMIETSRHISNKNKVQHNDWKDSPIGKMRILTRTVVPAGKSRSV<br>IIDKFETTLGQTLYIKEFKDNRIIDSSQKEQENKTQTIINKKDIHHSSFLIDSNNIQELLDNPLVGRISFKKVLKIIDLLINSDLSCSEVANKLNICEEV<br>IRLIKMQLR<br></div><h2>Table T2:</h2><table class="borderedTbl"><thead><tr><th>Template</th><th>Seq Identity</th><th>Oligo-state</th><th>QSQE</th><th>Found by</th><th>Method</th><th>Resolution</th><th>Seq Similarity</th><th>Coverage</th><th>Description</th></tr></thead><tbody><tr><td>3hfw.1.A</td><td>24.84</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.92&Aring;</td><td>0.30</td><td>0.63</td><td>Protein ADP-ribosylarginine hydrolase</td></tr><tr><td>6g28.1.A</td><td>24.92</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.23&Aring;</td><td>0.30</td><td>0.63</td><td>[Protein ADP-ribosylarginine] hydrolase</td></tr><tr><td>2fp0.1.A</td><td>18.83</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.05&Aring;</td><td>0.29</td><td>0.64</td><td>ADP-ribosylhydrolase like 2</td></tr><tr><td>6d3a.1.A</td><td>18.52</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.60&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>5zqy.1.A</td><td>18.52</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.58&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>6d36.1.A</td><td>18.65</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.70&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>6d36.2.A</td><td>18.65</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.70&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>6d3a.2.A</td><td>18.52</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.60&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>1t5j.1.A</td><td>25.08</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.70&Aring;</td><td>0.31</td><td>0.58</td><td>Hypothetical protein MJ1187</td></tr><tr><td>2qty.1.A</td><td>18.52</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.80&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>2qty.2.A</td><td>18.52</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.80&Aring;</td><td>0.29</td><td>0.64</td><td>Poly(ADP-ribose) glycohydrolase ARH3</td></tr><tr><td>2g4k.1.A</td><td>19.00</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.82&Aring;</td><td>0.29</td><td>0.63</td><td>ADP-ribosylhydrolase 3</td></tr><tr><td>6g1p.1.A</td><td>20.19</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.55&Aring;</td><td>0.29</td><td>0.63</td><td>ADP-ribosylhydrolase like 2</td></tr><tr><td>6dre.1.A</td><td>23.59</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.80&Aring;</td><td>0.32</td><td>0.56</td><td>ADP-ribosyl-(Dinitrogen reductase) hydrolase</td></tr><tr><td>2wod.1.A</td><td>22.06</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.25&Aring;</td><td>0.30</td><td>0.55</td><td>ADP-RIBOSYL-[DINITROGEN REDUCTASE] GLYCOHYDROLASE</td></tr><tr><td>2cwc.1.A</td><td>23.88</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.65&Aring;</td><td>0.31</td><td>0.57</td><td>ADP-ribosylglycohydrolase</td></tr><tr><td>3o5t.1.E</td><td>21.79</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.09&Aring;</td><td>0.30</td><td>0.55</td><td>Dinitrogenase reductase activacting glicohydrolase</td></tr><tr><td>3g9d.1.A</td><td>21.79</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.50&Aring;</td><td>0.30</td><td>0.55</td><td>Dinitrogenase reductase activating glucohydrolase</td></tr><tr><td>3g9d.2.A</td><td>21.79</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.50&Aring;</td><td>0.30</td><td>0.55</td><td>Dinitrogenase reductase activating glucohydrolase</td></tr><tr><td>2woe.1.A</td><td>22.06</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>1.90&Aring;</td><td>0.30</td><td>0.55</td><td>ADP-RIBOSYL-[DINITROGEN REDUCTASE] GLYCOHYDROLASE</td></tr><tr><td>2woc.1.A</td><td>22.06</td><td>monomer</td><td></td><td>HHblits</td><td>X-ray</td><td>2.20&Aring;</td><td>0.30</td><td>0.55</td><td>ADP-RIBOSYL-[DINITROGEN REDUCTASE] GLYCOHYDROLASE</td></tr><tr><td>1t5j.1.A</td><td>29.48</td><td>monomer</td><td></td><td>BLAST</td><td>X-ray</td><td>2.70&Aring;</td><td>0.34</td><td>0.53</td><td>Hypothetical protein MJ1187</td></tr><tr><td>6dre.1.A</td><td>26.19</td><td>monomer</td><td></td><td>BLAST</td><td>X-ray</td><td>1.80&Aring;</td><td>0.35</td><td>0.33</td><td>ADP-ribosyl-(Dinitrogen reductase) hydrolase</td></tr><tr><td>2woc.1.A</td><td>32.89</td><td>monomer</td><td></td><td>BLAST</td><td>X-ray</td><td>2.20&Aring;</td><td>0.36</td><td>0.30</td><td>ADP-RIBOSYL-[DINITROGEN REDUCTASE] GLYCOHYDROLASE</td></tr><tr><td>2wod.1.A</td><td>32.89</td><td>monomer</td><td></td><td>BLAST</td><td>X-ray</td><td>2.25&Aring;</td><td>0.36</td><td>0.30</td><td>ADP-RIBOSYL-[DINITROGEN REDUCTASE] GLYCOHYDROLASE</td></tr><tr><td>2woe.1.A</td><td>32.89</td><td>monomer</td><td></td><td>BLAST</td><td>X-ray</td><td>1.90&Aring;</td><td>0.36</td><td>0.30</td><td>ADP-RIBOSYL-[DINITROGEN REDUCTASE] GLYCOHYDROLASE</td></tr><tr><td>2jrt.1.A</td><td>13.33</td><td>monomer</td><td></td><td>HHblits</td><td>NMR</td><td>NA</td><td>0.27</td><td>0.09</td><td>Uncharacterized protein</td></tr><tr><td>1oyi.1.A</td><td>11.76</td><td>monomer</td><td></td><td>HHblits</td><td>NMR</td><td>NA</td><td>0.28</td><td>0.07</td><td>double-stranded RNA-binding protein</td></tr></tbody></table></div><div id="sib_footer"><div id="sib_footer_content"><a href="https://www.sib.swiss">Swiss Institute of Bioinformatics</a><a href="mailto:help-swissmodel@unibas.ch" style="margin-left:16px;">Contact Us</a><div id="sib_footer_right" style="float:right"></div></div></div></body></html>
